pkg/apis/core/v1/defaults.go

```diff
@@ -203,6 +205,11 @@ func SetDefaults_PodSpec(obj *v1.PodSpec) {
 	if obj.SchedulerName == "" {
 		obj.SchedulerName = v1.DefaultSchedulerName
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.WorkloadInfoDefaulting) {
+		if len(obj.WorkloadInfo) == 0 {
+			_ = obj.Workloads()
```

Comment:
in the admission controller for vertical scaling you added a few weeks ago, there is already a setter level function in the types.go file for this purpose,  suggested to make this consistent, i.e. for workloadinfo slice life cycle: init(), set() in the admission controller, and get()afterward in other Arktos code except the admission controller ( when the pod is first time created in the system ). likely a more readable and maintainable code as such.

Reply:
This is not set. We are calling get here to allocate and initialize WorkloadInfo for first time as defaulting.